ext/src/ruby_api/engine.rs

```diff
@@ -85,7 +88,7 @@ impl Engine {
 
                 EngineImpl::new(&config).map_err(|e| error!("{}", e))?
             }
-            None => EngineImpl::default(),
+            None => EngineImpl::new(&default_config()).map_err(|e| error!("{}", e))?,
```

Comment:
> we keep the current new functionality flexible in the sense that you either pass some configuration options or get the real default back

I don't see a much of a use case for this particular option, unless I'm missing something. I've thought about this a good bit, and I think that reporting memory is the best possible default for users. Without it, we risk thrashing the Ruby GC and causing performance problems for the rest of their app.

Is there a specific use case where disabling memory reporting is desired? In that case, the user would probably be better off running Wasmtime in a separate service

Reply:
Sounds reasonable to me. I was just a bit confused, since the original state of the PR hinted that we might want to offer both options. Reading your comment here https://github.com/bytecodealliance/wasmtime-rb/pull/187#issuecomment-1543037806 I understand that this is not the case, so we're aligned on this.